contracts/triggers/AaveQuotePriceTrigger.sol

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity =0.8.10;
+
+import "../interfaces/ITrigger.sol";
+import "../interfaces/aaveV3/IAaveV3Oracle.sol";
+import "../interfaces/lido/IWStEth.sol";
+import "../auth/AdminAuth.sol";
+import "../DS/DSMath.sol";
+import "../utils/TokenUtils.sol";
+import "../actions/aaveV3/helpers/AaveV3RatioHelper.sol";
+
+/// @title Trigger contract that verifies if current token price ratio is over/under the price ratio specified during subscription
+contract AaveQuotePriceTrigger is ITrigger, AdminAuth, DSMath, AaveV3RatioHelper {
+    using TokenUtils for address;
+
+    enum PriceState {
+        OVER,
+        UNDER
+    }
+
+    /// @param baseTokenAddr address of the base token which is quoted
+    /// @param quoteTokenAddr address of the quote token
+    /// @param price price in quote token of the base token that represents the triggerable point
+    /// @param state represents if we want the current price to be higher or lower than price param
+    struct SubParams {
+        address baseTokenAddr;
+        address quoteTokenAddr;
+        uint256 price;
+        uint8 state;
+    }
+
+    IAaveV3Oracle public constant aaveOracleV3 =
+        IAaveV3Oracle(AAVE_ORACLE_V3);
+
+    /// @dev checks chainlink oracle for current prices and triggers if it's in a correct state
+    function isTriggered(bytes memory, bytes memory _subData) public view override returns (bool) {
+        SubParams memory triggerSubData = parseSubInputs(_subData);
+
+
+        uint256 currPrice = getPrice(triggerSubData.baseTokenAddr, triggerSubData.quoteTokenAddr);
+
+        if (PriceState(triggerSubData.state) == PriceState.OVER) {
+            if (currPrice > triggerSubData.price) return true;
+        }
+
+        if (PriceState(triggerSubData.state) == PriceState.UNDER) {
+            if (currPrice < triggerSubData.price) return true;
+        }
+
+        return false;
+    }
+
+    /// @dev helper function that returns latest base token price in quote tokens
+    function getPrice(address _baseTokenAddr, address _quoteTokenAddr) public view returns (uint256 price) {
+        address baseTokenAddr = _baseTokenAddr;
+        if (baseTokenAddr == TokenUtils.WETH_ADDR) {
+            baseTokenAddr = TokenUtils.ETH_ADDR;
+        } else if (baseTokenAddr == TokenUtils.WSTETH_ADDR) {
+            baseTokenAddr = TokenUtils.STETH_ADDR;
+        }
+
+        address quoteTokenAddr = _quoteTokenAddr;
+        if (quoteTokenAddr == TokenUtils.WETH_ADDR) {
+            quoteTokenAddr = TokenUtils.ETH_ADDR;
+        } else if (quoteTokenAddr == TokenUtils.WSTETH_ADDR) {
+            quoteTokenAddr = TokenUtils.STETH_ADDR;
+        }
+
+        address[] memory assets = new address[](2);
+        assets[0] = _baseTokenAddr;
```

Comment:
This should be `baseTokenAddr` and `quoteTokenAddr` maybe better naming to avoid this kind of confusion